java/org/apache/coyote/AbstractProtocol.java

```diff
@@ -347,22 +352,27 @@ public String getName() {
     private String getNameInternal() {
         StringBuilder name = new StringBuilder(getNamePrefix());
         name.append('-');
-        if (getAddress() != null) {
-            name.append(getAddress().getHostAddress());
-            name.append('-');
+        if (getPath() != null) {
+            name.append(getPath().getFileName().toString());
```

Comment:
Stupid question: What will happen when I bind two connectors on two different directory with the same socket file name?

Reply:
In theory the name will clash.

We could try the full path separated by dashes? How do we handle windows though, ignore all special characters?

Reply:
I need to think about this, ugly...

Reply:
I think we should use the full path and I will explain why. The name contains the addresss where the socket is bound. TCP sockets contains IP and port which makes then fully idenfiable. UDS have they full path. I would use it, not think about it. Windows uses a special path for UDS.

Reply:
Agreed - fixed.